### PR TITLE
Keep date format after reshaping

### PR DESCRIPTION
**Pull request by etiennebacher**
Before:
``` r
family <- tidyr::tibble(
  family = 1:3,
  dob_child1 = as.Date(c("1998-11-26", "2004-10-10", "2000-12-05")),
  dob_child2 = as.Date(c("2000-01-29", NA, "2004-04-05"))
)
family
#> # A tibble: 3 × 3
#>   family dob_child1 dob_child2
#>    <int> <date>     <date>    
#> 1      1 1998-11-26 2000-01-29
#> 2      2 2004-10-10 NA        
#> 3      3 2000-12-05 2004-04-05

tidyr::pivot_longer(family, !family, names_to = "child")
#> # A tibble: 6 × 3
#>   family child      value     
#>    <int> <chr>      <date>    
#> 1      1 dob_child1 1998-11-26
#> 2      1 dob_child2 2000-01-29
#> 3      2 dob_child1 2004-10-10
#> 4      2 dob_child2 NA        
#> 5      3 dob_child1 2000-12-05
#> 6      3 dob_child2 2004-04-05
datawizard::data_to_long(family, -c("family"), names_to = "child")
#> # A tibble: 6 × 3
#>   family child      value
#>    <int> <chr>      <dbl>
#> 1      1 dob_child1 10556
#> 2      1 dob_child2 10985
#> 3      2 dob_child1 12701
#> 4      2 dob_child2    NA
#> 5      3 dob_child1 11296
#> 6      3 dob_child2 12513
```

After:
``` r
family <- tidyr::tibble(
  family = 1:3,
  dob_child1 = as.Date(c("1998-11-26", "2004-10-10", "2000-12-05")),
  dob_child2 = as.Date(c("2000-01-29", NA, "2004-04-05"))
)
family
#> # A tibble: 3 × 3
#>   family dob_child1 dob_child2
#>    <int> <date>     <date>    
#> 1      1 1998-11-26 2000-01-29
#> 2      2 2004-10-10 NA        
#> 3      3 2000-12-05 2004-04-05

tidyr::pivot_longer(family, !family, names_to = "child")
#> # A tibble: 6 × 3
#>   family child      value     
#>    <int> <chr>      <date>    
#> 1      1 dob_child1 1998-11-26
#> 2      1 dob_child2 2000-01-29
#> 3      2 dob_child1 2004-10-10
#> 4      2 dob_child2 NA        
#> 5      3 dob_child1 2000-12-05
#> 6      3 dob_child2 2004-04-05
datawizard::data_to_long(family, -c("family"), names_to = "child")
#> # A tibble: 6 × 3
#>   family child      value     
#>    <int> <chr>      <date>    
#> 1      1 dob_child1 1998-11-26
#> 2      1 dob_child2 2000-01-29
#> 3      2 dob_child1 2004-10-10
#> 4      2 dob_child2 NA        
#> 5      3 dob_child1 2000-12-05
#> 6      3 dob_child2 2004-04-05
```

### CODE REVIEW

File: R/data_to_long.R [hunk 1]
```diff
@@ -129,6 +129,19 @@ data_to_long <- function(data,
     )
   }
 
+  if (length(names_to) == 1) {
```
Comment (IndrajeetPatil) [inline R/data_to_long.R:132]:
```suggestion
  if (length(names_to) == 1L) {
```

File: R/data_to_wide.R [hunk 2]
```diff
@@ -258,6 +258,16 @@ data_to_wide <- function(data,
     attributes(out[[i]]) <- variable_attr[[i]]
   }
 
+  # convert back to date if original values were dates
+  values_are_dates <- all(vapply(data[, values_from, drop = FALSE], function(x) {
+    inherits(x, "Date")
+  }, FUN.VALUE = logical(1)))
```
Comment (IndrajeetPatil) [inline R/data_to_wide.R:264]: Nit: Extract out the anonymous function to a helper `.is_date()`, and use that instead.

Ideally, `apply()` calls should be one-lines to improve readability.

**Review by IndrajeetPatil — APPROVED** [review]

**Comment by etiennebacher** [comment]
@IndrajeetPatil  This fails on R 3.6, is it possible to cancel the merge?

**Comment by IndrajeetPatil** [comment]
I have reverted the change. Sorry I missed that.

**Comment by etiennebacher** [comment]
Thanks

**Pull request merged**